src/index.js

```diff
@@ -1,22 +1,22 @@
 // @flow
-type EqualityFn = (a: any, b: any) => boolean;
+type EqualityFn = (a: mixed, b: mixed) => boolean;
 
-const simpleIsEqual = (a: any, b: any): boolean => a === b;
+const simpleIsEqual: EqualityFn = (a: mixed, b: mixed): boolean => a === b;
 
-export default function (resultFn: Function, isEqual?: EqualityFn = simpleIsEqual) {
-  let lastThis: any;
-  let lastArgs: Array<any> = [];
-  let lastResult: any;
+export default function <ResultFn: Function>(resultFn: ResultFn, isEqual?: EqualityFn = simpleIsEqual): ResultFn {
```

Comment:
Hmm, weird... that used to work. Maybe [recent changes to arity](https://flow.org/blog/2017/05/07/Strict-Function-Call-Arity/) have changed that.

You could make it a little bit stronger by using:

```js
(...Array<any>) => mixed
```